tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -67,12 +50,47 @@ uint32_t DeclResultIdMapper::getDeclResultId(const NamedDecl *decl) const {
   return 0;
 }
 
-uint32_t DeclResultIdMapper::getOrRegisterDeclResultId(const NamedDecl *decl) {
-  if (const auto *info = getDeclSpirvInfo(decl))
+uint32_t DeclResultIdMapper::createFnParam(uint32_t paramType,
+                                           const ParmVarDecl *param) {
+  const uint32_t id = theBuilder.addFnParam(paramType, param->getName());
+  astDecls[param] = {id, spv::StorageClass::Function};
+
+  return id;
+}
+
+uint32_t DeclResultIdMapper::createFnVar(uint32_t varType, const VarDecl *var,
+                                         llvm::Optional<uint32_t> init) {
+  const uint32_t id = theBuilder.addFnVar(varType, var->getName(), init);
+  astDecls[var] = {id, spv::StorageClass::Function};
+
+  return id;
+}
+
+uint32_t DeclResultIdMapper::createFileVar(uint32_t varType, const VarDecl *var,
```

Comment:
What's the difference between FileVar and ModuleVar?
If there's no difference, I'd suggest renaming this function to `createModuleVar` to be consistent.